src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -183,13 +203,52 @@ private SymbolicRegexMatcher(SymbolicRegexBuilder<TSet> builder, SymbolicRegexNo
                 _positionKinds[mintermId + 1] = CalculateMintermIdKind(mintermId);
             }
 
+            // Create optimized reversal
+            _optimizedReversalState = CreateOptimizedReversal(_pattern.Reverse(builder));
+
             // Store the find optimizations that can be used to jump ahead to the next possible starting location.
             // If there's a leading beginning anchor, the find optimizations are unnecessary on top of the DFA's
             // handling for beginning anchors.
             if (findOptimizations.IsUseful &&
                 findOptimizations.LeadingAnchor is not RegexNodeKind.Beginning)
             {
-                _findOpts = findOptimizations;
+                // this makes some assumptions about the frequency of occurrences
+                // some large sets like \p{Sm} are faster with infrequent matches but slower with frequent matches
+                // the easiest thing to do here is to leave it as-is, but this means some inputs can have large performance losses of 10x or more
+
+                var setIsTooCommon = new Func<RegexFindOptimizations.FixedDistanceSet, bool>((fds) =>
+                {
+                    // _wout($"rn{fds.Range is null}");
+                    // _wout($"cn{fds.Chars is null}");
+                    // _wout($"cc{fds.Chars!.Length}");
+                    return fds switch
+                    {
+                        { Chars: not null } =>
+                            // anything above 4 uint16 chars is generally slower than DFA
+                            fds.Negated ||
+                            (fds.Chars.Length > 4 &&
+                            Array.Exists(fds.Chars, char.IsAsciiLetterLower)),
```

Comment:
This heuristic seems like it'd be very specific to the data set, no?

Reply:
This should definitely be changed, i'm still thinking of how to decide on this.

this should really account for two things:
- How difficult it is to search for the set (number of chars and chosen SearchValues<char> implementation)
- How rare the set is (hard to make good assumptions but something like uppercase characters being more rare than lowercase characters)